src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py

```diff
@@ -1694,6 +1718,96 @@ def _get_or_add_extension(extension_name, extension_module, update=False):
     return True
 
 
+def _ensure_default_log_analytics_workspace_for_monitoring(cmd, subscription_id, resource_group_name):
+    # log analytics workspaces cannot be created in WCUS region due to capacity limits so mapped to EUS per discussion with log analytics team
+    AzureLocationToOmsRegionCodeMap = {
+        "eastus": "EUS",
+        "westeurope": "WEU",
+        "southeastasia": "SEA",
+        "australiasoutheast": "ASE",
+        "usgovvirginia": "USGV",
+        "westcentralus": "EUS",
+        "japaneast": "EJP",
+        "uksouth": "SUK",
+        "canadacentral": "CCA",
+        "centralindia": "CIN",
+        "eastus2euap": "EAP"
+    }
+    AzureRegionToOmsRegionMap = {
+        "australiaeast": "australiasoutheast",
+        "australiasoutheast": "australiasoutheast",
+        "brazilsouth": "eastus",
+        "canadacentral": "canadacentral",
+        "canadaeast": "canadacentral",
+        "centralus": "eastus",
+        "eastasia": "southeastasia",
+        "eastus": "eastus",
+        "eastus2": "eastus",
+        "japaneast": "japaneast",
+        "japanwest": "japaneast",
+        "northcentralus": "eastus",
+        "northeurope": "westeurope",
+        "southcentralus": "eastus",
+        "southeastasia": "southeastasia",
+        "uksouth": "uksouth",
+        "ukwest": "uksouth",
+        "westcentralus": "eastus",
+        "westeurope": "westeurope",
+        "westus": "eastus",
+        "westus2": "eastus",
+        "centralindia": "centralindia",
+        "southindia": "centralindia",
+        "westindia": "centralindia",
+        "koreacentral": "southeastasia",
+        "koreasouth": "southeastasia",
+        "francecentral": "westeurope",
+        "francesouth": "westeurope"
+    }
+
+    rg_location = _get_rg_location(cmd.cli_ctx, resource_group_name)
+    default_region_name = "eastus"
+    default_region_code = "EUS"
+
+    workspace_region = AzureRegionToOmsRegionMap[rg_location] if AzureRegionToOmsRegionMap[rg_location] else default_region_name
+    workspace_regionCode = AzureLocationToOmsRegionCodeMap[workspace_region] if AzureLocationToOmsRegionCodeMap[workspace_region] else default_region_code
+
+    default_workspace_resource_group = 'DefaultResourceGroup-' + workspace_regionCode
+    default_workspace_name = 'DefaultWorkspace-{0}-{1}'.format(subscription_id, workspace_regionCode)
+
+    default_workspace_resource_Id = '/subscriptions/{0}/resourceGroups/{1}/providers/Microsoft.OperationalInsights/workspaces/{2}'.format(subscription_id, default_workspace_resource_group, default_workspace_name)
+    resource_groups = cf_resource_groups(cmd.cli_ctx, subscription_id)
```

Comment:
Question:  is subscription_id the one from the workspace or the cluster? I'm thinking about the case where the workspace is in a different subscription.

Reply:
Talked with Ganga, since this is only being used for the default workspace which is always in the subscription the cluster is in. We don't need to update the id.